frontend/src/components/input.tsx

```diff
@@ -0,0 +1,43 @@
+import React, { ChangeEvent, FormEvent, useCallback, useMemo, useState } from 'react';
+import styles from './input.module.css';
+import RightArrow from '../icons/map-arrow-right.svg';
+import classNames from 'classnames';
+
+export interface InputProps {
+  sendMessage: (message: string) => void;
+}
+
+export const Input = ({ sendMessage }: InputProps) => {
+  const [userInut, setUserInput] = useState<string>('');
+
+  const onChange = useCallback((event: ChangeEvent<HTMLInputElement>) => {
+    setUserInput(event.target.value);
+  }, []);
+
+  const onSend = useCallback(
+    (event: FormEvent<HTMLElement>) => {
+      event.preventDefault();
+      sendMessage(userInut);
+      setUserInput('');
+    },
+    [sendMessage, userInut]
+  );
+
+  const sendDisabled = useMemo(() => userInut.length === 0, [userInut]);
```

Comment:
What does useMemo do?

Reply:
It is a bit of overkill here, but it is an optimization method that utilizes caching. The variable, `sendDisabled` , will only get recalculated if any of the dependencies change, in this case just `userInput`.

It would be the same to just have `const sendDisabled = userInput.length === 0` , but this will re-run every time the component is re-rendered